src/io.cc

```diff
@@ -100,23 +85,22 @@ IO::close()
 /**
  * to_string
  *
- * Copies up to n characters from the current stream position into the
- * destination buffer. Adds null terminator at the end of the copied string.
+ * Reads up to n characters from the current stream position into a string.
+ * Reading stops if either n characters have been read or end of file is
+ * reached.
  *
- * @param dest The destination buffer to copy characters into
- * @param n The maximum number of characters to copy
- * @return void
+ * @param n The maximum number of characters to read
+ * @return std::string containing the read characters
  */
-void
-IO::to_string(char* dest, std::size_t n)
-{
-    std::size_t i = 0;
+std::string IO::to_string(std::size_t n) {
+    std::string result;
+    result.reserve(n);
     while (n > 0 && !eof()) {
-        dest[i++] = current_char_;
+        result += current_char_;
         next();
```

Comment:
an idiomatic `next` in C++
template <typename Iter>
std::string::iterator next(std::string::iterator iter)
{
    return ++iter;
}